test/known_problems/should_pass/guard_negative_refinement_should_pass.erl

```diff
@@ -0,0 +1,15 @@
+-module(guard_negative_refinement_should_pass).
+
+-export([i/1]).
+
+-type state() :: #{value := integer() | nil}.
+
+-spec i(state()) -> integer().
+i(State) when map_get(value, State) /= nil ->
```

Comment:
This guard can fail. Shouldn't we have another clause for value nil? (Not that we can detect this non-exhaustiveness now, but the code makes more sense that way and maybe in the future we can check it.)

Reply:
Good point, I've added a clause for `nil` (d3b5fa06).

Reply:
Fixed: 0c9307cc. I've also added `nil` cases for the other functions (2b2c5110).